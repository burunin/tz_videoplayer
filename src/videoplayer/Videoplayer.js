import React, { Component } from "react";
import source from "../../src/assets/video.mp4";
import "./Videoplayer.scss";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

export default class Videoplayer extends Component {
  constructor(props) {
    super(props);
    this.state = { mediaTime: 0, progress: 0, currentTime: 0, paused: false };
    this.video = React.createRef();
    this.timerWrapper = React.createRef();
    this.intervalFwd = undefined;
    this.intervalRwd = undefined;
  }

  componentDidMount() {
    return this.video.current.canplay ? this.play() : 0;
  }

  stopMedia(e) {
    this.pause();
    this.video.current.currentTime = 0;
    clearInterval(this.intervalRwd);
    clearInterval(this.intervalFwd);
    return e ? e.stopPropagation() : 0;
  }

  play() {
    this.setState({ paused: false });
    return this.video.current.play();
  }

  pause() {
    this.setState({ paused: true });
    return this.video.current.pause();
  }

  playPauseMedia() {
    clearInterval(this.intervalRwd);
    clearInterval(this.intervalFwd);
    if (this.video.current.paused) {
      this.video.current.play();
      this.setState({
        paused: false
      });
    } else {
      this.video.current.pause();
      this.setState({
        paused: true
      });
    }
  }

  mediaBackward() {
    clearInterval(this.intervalFwd);
    clearInterval(this.intervalRwd);
    this.pause();
    this.intervalRwd = setInterval(
      () => this.windBackward(this.video.current),
      200
    );
  }

  mediaForward() {
    clearInterval(this.intervalFwd);
    clearInterval(this.intervalRwd);
    this.pause();
    this.intervalFwd = setInterval(
      () => this.windForward(this.video.current),
      200
    );
  }

  windBackward(video) {
    if (video.currentTime <= 2) {
      this.stopMedia();
    } else {
      video.currentTime -= 2;
    }
  }

  windForward(video) {
    if (video.currentTime >= video.duration - 2) {
      this.stopMedia();
    } else {
      video.currentTime += 2;
    }
  }

  setTime() {
    var minutes = Math.floor(this.video.current.currentTime / 60);
    var seconds = Math.floor(this.video.current.currentTime - minutes * 60);
    var minuteValue;
    var secondValue;

    if (minutes < 10) {
      minuteValue = "0" + minutes;
    } else {
      minuteValue = minutes;
    }

    if (seconds < 10) {
      secondValue = "0" + seconds;
    } else {
      secondValue = seconds;
    }

    this.setState({
      mediaTime: minuteValue + ":" + secondValue,
      progressBarLength:
        this.timerWrapper.current.clientWidth *
        (this.video.current.currentTime / this.video.current.duration)
    });
  }

  render() {
    const progressWidth = { width: this.state.progressBarLength + "px" };

    return (
      <div className="player">
        <div className="video-container">
          <video
            autoPlay={true}
            className="video-screen"
            ref={this.video}
            onTimeUpdate={() => this.setTime()}
            onEnded={() => this.stopMedia()}
          >
            <source src={source} type="video/mp4" />
          </video>
        </div>
        <div className="controls">
          <div
            className="control-icon play"
            onClick={() => this.playPauseMedia()}
          >
            {this.state.paused ? (
              <FontAwesomeIcon
                icon="play"
                size="2x"
                className="control-icon play"
              />
            ) : (
              <FontAwesomeIcon
                icon="pause"
                size="2x"
                className="control-icon pause"
              />
            )}
          </div>
          <div className="control-icon stop" onClick={e => this.stopMedia(e)}>
            <FontAwesomeIcon icon="stop" size="2x" className="stop" />
          </div>
          <div
            className="control-icon rwd"
            onClick={() => this.mediaBackward()}
          >
            <FontAwesomeIcon icon="fast-backward" size="2x" className="stop" />
          </div>
          <div className="control-icon fwd" onClick={() => this.mediaForward()}>
            <FontAwesomeIcon icon="fast-forward" size="2x" className="stop" />
          </div>
          <div className="timer" ref={this.timerWrapper}>
            <div style={progressWidth} />
            <span>{this.state.mediaTime}</span>
          </div>
        </div>
      </div>
    );
  }
}
