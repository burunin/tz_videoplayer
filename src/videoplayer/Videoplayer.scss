.control-icons {
  &.stop {
    color: white;
  }
}
