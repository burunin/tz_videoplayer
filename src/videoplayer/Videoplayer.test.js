import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount } from "enzyme";
import Videoplayer from "./Videoplayer";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

import { library } from "@fortawesome/fontawesome-svg-core";
import {
  faStop,
  faPlay,
  faPause,
  faFastForward,
  faFastBackward
} from "@fortawesome/free-solid-svg-icons";

library.add(faStop, faPlay, faPause, faFastForward, faFastBackward);

describe("App", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Videoplayer />);
  });

  test("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should have the `video` element", () => {
    expect(
      wrapper.containsMatchingElement(
        <video>
          <source />
        </video>
      )
    ).toBe(true);
  });

  test("should have a `source` element", () => {
    expect(wrapper.containsMatchingElement(<source />)).toBe(true);
  });

  it("should render source element with src attribute", () => {
    const source = wrapper.find("source");
    expect(source.props()).toHaveProperty("src");
  });

  test("pass video events stopMedia", () => {
    wrapper.instance().stopMedia = jest.fn().mockReturnValue(1);
    wrapper.find(".video-screen").simulate("ended");
    expect(wrapper.instance().stopMedia).toHaveBeenCalled();
  });

  describe("Test if controlls working", () => {
    test("click play/pause", () => {
      wrapper.instance().playPauseMedia = jest.fn().mockReturnValue(0);
      wrapper.find(".control-icon.play").simulate("click");
      expect(wrapper.instance().playPauseMedia).toReturn();
    });
    test("click stop", () => {
      wrapper.instance().stopMedia = jest.fn().mockReturnValue(0);
      wrapper.find(".control-icon.stop").simulate("click");
      expect(wrapper.instance().stopMedia).toReturn();
    });

    test("click rewind", () => {
      wrapper.instance().mediaBackward = jest.fn().mockReturnValue(0);
      wrapper.find(".control-icon.rwd").simulate("click");
      expect(wrapper.instance().mediaBackward).toReturn();
    });

    test("click fast-forward", () => {
      wrapper.instance().mediaForward = jest.fn().mockReturnValue(0);
      wrapper.find(".control-icon.fwd").simulate("click");
      expect(wrapper.instance().mediaForward).toReturn();
    });
  });

  test("pass video events setTime", () => {
    wrapper.instance().setTime = jest.fn().mockReturnValue(1);
    wrapper.find(".video-screen").simulate("timeUpdate");
    expect(wrapper.instance().setTime).toHaveBeenCalled();
  });

  describe("test methods", () => {
    test("check stopMedia", () => {
      wrapper = mount(<Videoplayer />);
      wrapper.instance().pause = jest.fn();
      const e = { stopPropagation: jest.fn().mockReturnValue(1) };
      wrapper.find(".control-icon.stop").simulate("click");
      expect(wrapper.instance().pause).toHaveBeenCalled();
      expect(wrapper.instance().stopMedia()).toBe(0);
      expect(wrapper.instance().stopMedia(e)).toBe(1);
    });

    test("check play", () => {
      wrapper = mount(<Videoplayer />);
      wrapper.instance().video.current.play = jest.fn();
      wrapper.find(".control-icon.play").simulate("click");
      expect(wrapper.instance().video.current.play).toHaveBeenCalled();
      expect(wrapper.state().paused).toBe(false);
    });

    test("check pause", () => {
      wrapper = mount(<Videoplayer />);
      wrapper.instance().video.current.pause = jest.fn();
      wrapper.find(".stop.control-icon").simulate("click");
      expect(wrapper.instance().video.current.pause).toHaveBeenCalled();
      expect(wrapper.state().paused).toBe(true);
    });

    test("check pausePlayMedia1", () => {});
    test("check pausePlayMedia2", () => {});
  });
});
