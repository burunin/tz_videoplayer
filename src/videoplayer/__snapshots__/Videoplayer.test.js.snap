// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Videoplayer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="video-container"
        >
          <video
            autoPlay={true}
            className="video-screen"
            onEnded={[Function]}
            onTimeUpdate={[Function]}
          >
            <source
              src="video.mp4"
              type="video/mp4"
            />
          </video>
        </div>,
        <div
          className="controls"
        >
          <div
            className="control-icon play"
            onClick={[Function]}
          >
            <FontAwesomeIcon
              border={false}
              className="control-icon pause"
              fixedWidth={false}
              flip={null}
              icon="pause"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size="2x"
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
          </div>
          <div
            className="control-icon stop"
            onClick={[Function]}
          >
            <FontAwesomeIcon
              border={false}
              className="stop"
              fixedWidth={false}
              flip={null}
              icon="stop"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size="2x"
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
          </div>
          <div
            className="control-icon rwd"
            onClick={[Function]}
          >
            <FontAwesomeIcon
              border={false}
              className="stop"
              fixedWidth={false}
              flip={null}
              icon="fast-backward"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size="2x"
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
          </div>
          <div
            className="control-icon fwd"
            onClick={[Function]}
          >
            <FontAwesomeIcon
              border={false}
              className="stop"
              fixedWidth={false}
              flip={null}
              icon="fast-forward"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size="2x"
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
          </div>
          <div
            className="timer"
          >
            <div
              style={
                Object {
                  "width": "undefinedpx",
                }
              }
            />
            <span />
          </div>
        </div>,
      ],
      "className": "player",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <video
            autoPlay={true}
            className="video-screen"
            onEnded={[Function]}
            onTimeUpdate={[Function]}
          >
            <source
              src="video.mp4"
              type="video/mp4"
            />
          </video>,
          "className": "video-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoPlay": true,
            "children": <source
              src="video.mp4"
              type="video/mp4"
            />,
            "className": "video-screen",
            "onEnded": [Function],
            "onTimeUpdate": [Function],
          },
          "ref": Object {
            "current": null,
          },
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": "video.mp4",
              "type": "video/mp4",
            },
            "ref": null,
            "rendered": null,
            "type": "source",
          },
          "type": "video",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="control-icon play"
              onClick={[Function]}
            >
              <FontAwesomeIcon
                border={false}
                className="control-icon pause"
                fixedWidth={false}
                flip={null}
                icon="pause"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="2x"
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </div>,
            <div
              className="control-icon stop"
              onClick={[Function]}
            >
              <FontAwesomeIcon
                border={false}
                className="stop"
                fixedWidth={false}
                flip={null}
                icon="stop"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="2x"
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </div>,
            <div
              className="control-icon rwd"
              onClick={[Function]}
            >
              <FontAwesomeIcon
                border={false}
                className="stop"
                fixedWidth={false}
                flip={null}
                icon="fast-backward"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="2x"
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </div>,
            <div
              className="control-icon fwd"
              onClick={[Function]}
            >
              <FontAwesomeIcon
                border={false}
                className="stop"
                fixedWidth={false}
                flip={null}
                icon="fast-forward"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="2x"
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </div>,
            <div
              className="timer"
            >
              <div
                style={
                  Object {
                    "width": "undefinedpx",
                  }
                }
              />
              <span />
            </div>,
          ],
          "className": "controls",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FontAwesomeIcon
                border={false}
                className="control-icon pause"
                fixedWidth={false}
                flip={null}
                icon="pause"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="2x"
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />,
              "className": "control-icon play",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "border": false,
                "className": "control-icon pause",
                "fixedWidth": false,
                "flip": null,
                "icon": "pause",
                "inverse": false,
                "listItem": false,
                "mask": null,
                "pull": null,
                "pulse": false,
                "rotation": null,
                "size": "2x",
                "spin": false,
                "symbol": false,
                "title": "",
                "transform": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FontAwesomeIcon
                border={false}
                className="stop"
                fixedWidth={false}
                flip={null}
                icon="stop"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="2x"
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />,
              "className": "control-icon stop",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "border": false,
                "className": "stop",
                "fixedWidth": false,
                "flip": null,
                "icon": "stop",
                "inverse": false,
                "listItem": false,
                "mask": null,
                "pull": null,
                "pulse": false,
                "rotation": null,
                "size": "2x",
                "spin": false,
                "symbol": false,
                "title": "",
                "transform": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FontAwesomeIcon
                border={false}
                className="stop"
                fixedWidth={false}
                flip={null}
                icon="fast-backward"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="2x"
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />,
              "className": "control-icon rwd",
              "onClick": [Function],
            },
            "ref": Object {
              "current": null,
            },
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "border": false,
                "className": "stop",
                "fixedWidth": false,
                "flip": null,
                "icon": "fast-backward",
                "inverse": false,
                "listItem": false,
                "mask": null,
                "pull": null,
                "pulse": false,
                "rotation": null,
                "size": "2x",
                "spin": false,
                "symbol": false,
                "title": "",
                "transform": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FontAwesomeIcon
                border={false}
                className="stop"
                fixedWidth={false}
                flip={null}
                icon="fast-forward"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="2x"
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />,
              "className": "control-icon fwd",
              "onClick": [Function],
            },
            "ref": Object {
              "current": null,
            },
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "border": false,
                "className": "stop",
                "fixedWidth": false,
                "flip": null,
                "icon": "fast-forward",
                "inverse": false,
                "listItem": false,
                "mask": null,
                "pull": null,
                "pulse": false,
                "rotation": null,
                "size": "2x",
                "spin": false,
                "symbol": false,
                "title": "",
                "transform": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  style={
                    Object {
                      "width": "undefinedpx",
                    }
                  }
                />,
                <span />,
              ],
              "className": "timer",
            },
            "ref": Object {
              "current": null,
            },
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "style": Object {
                    "width": "undefinedpx",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="video-container"
          >
            <video
              autoPlay={true}
              className="video-screen"
              onEnded={[Function]}
              onTimeUpdate={[Function]}
            >
              <source
                src="video.mp4"
                type="video/mp4"
              />
            </video>
          </div>,
          <div
            className="controls"
          >
            <div
              className="control-icon play"
              onClick={[Function]}
            >
              <FontAwesomeIcon
                border={false}
                className="control-icon pause"
                fixedWidth={false}
                flip={null}
                icon="pause"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="2x"
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </div>
            <div
              className="control-icon stop"
              onClick={[Function]}
            >
              <FontAwesomeIcon
                border={false}
                className="stop"
                fixedWidth={false}
                flip={null}
                icon="stop"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="2x"
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </div>
            <div
              className="control-icon rwd"
              onClick={[Function]}
            >
              <FontAwesomeIcon
                border={false}
                className="stop"
                fixedWidth={false}
                flip={null}
                icon="fast-backward"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="2x"
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </div>
            <div
              className="control-icon fwd"
              onClick={[Function]}
            >
              <FontAwesomeIcon
                border={false}
                className="stop"
                fixedWidth={false}
                flip={null}
                icon="fast-forward"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="2x"
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </div>
            <div
              className="timer"
            >
              <div
                style={
                  Object {
                    "width": "undefinedpx",
                  }
                }
              />
              <span />
            </div>
          </div>,
        ],
        "className": "player",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <video
              autoPlay={true}
              className="video-screen"
              onEnded={[Function]}
              onTimeUpdate={[Function]}
            >
              <source
                src="video.mp4"
                type="video/mp4"
              />
            </video>,
            "className": "video-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoPlay": true,
              "children": <source
                src="video.mp4"
                type="video/mp4"
              />,
              "className": "video-screen",
              "onEnded": [Function],
              "onTimeUpdate": [Function],
            },
            "ref": Object {
              "current": null,
            },
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "src": "video.mp4",
                "type": "video/mp4",
              },
              "ref": null,
              "rendered": null,
              "type": "source",
            },
            "type": "video",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="control-icon play"
                onClick={[Function]}
              >
                <FontAwesomeIcon
                  border={false}
                  className="control-icon pause"
                  fixedWidth={false}
                  flip={null}
                  icon="pause"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size="2x"
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
              </div>,
              <div
                className="control-icon stop"
                onClick={[Function]}
              >
                <FontAwesomeIcon
                  border={false}
                  className="stop"
                  fixedWidth={false}
                  flip={null}
                  icon="stop"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size="2x"
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
              </div>,
              <div
                className="control-icon rwd"
                onClick={[Function]}
              >
                <FontAwesomeIcon
                  border={false}
                  className="stop"
                  fixedWidth={false}
                  flip={null}
                  icon="fast-backward"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size="2x"
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
              </div>,
              <div
                className="control-icon fwd"
                onClick={[Function]}
              >
                <FontAwesomeIcon
                  border={false}
                  className="stop"
                  fixedWidth={false}
                  flip={null}
                  icon="fast-forward"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size="2x"
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
              </div>,
              <div
                className="timer"
              >
                <div
                  style={
                    Object {
                      "width": "undefinedpx",
                    }
                  }
                />
                <span />
              </div>,
            ],
            "className": "controls",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <FontAwesomeIcon
                  border={false}
                  className="control-icon pause"
                  fixedWidth={false}
                  flip={null}
                  icon="pause"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size="2x"
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />,
                "className": "control-icon play",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "control-icon pause",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": "pause",
                  "inverse": false,
                  "listItem": false,
                  "mask": null,
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": "2x",
                  "spin": false,
                  "symbol": false,
                  "title": "",
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <FontAwesomeIcon
                  border={false}
                  className="stop"
                  fixedWidth={false}
                  flip={null}
                  icon="stop"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size="2x"
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />,
                "className": "control-icon stop",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "stop",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": "stop",
                  "inverse": false,
                  "listItem": false,
                  "mask": null,
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": "2x",
                  "spin": false,
                  "symbol": false,
                  "title": "",
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <FontAwesomeIcon
                  border={false}
                  className="stop"
                  fixedWidth={false}
                  flip={null}
                  icon="fast-backward"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size="2x"
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />,
                "className": "control-icon rwd",
                "onClick": [Function],
              },
              "ref": Object {
                "current": null,
              },
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "stop",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": "fast-backward",
                  "inverse": false,
                  "listItem": false,
                  "mask": null,
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": "2x",
                  "spin": false,
                  "symbol": false,
                  "title": "",
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <FontAwesomeIcon
                  border={false}
                  className="stop"
                  fixedWidth={false}
                  flip={null}
                  icon="fast-forward"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size="2x"
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />,
                "className": "control-icon fwd",
                "onClick": [Function],
              },
              "ref": Object {
                "current": null,
              },
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "stop",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": "fast-forward",
                  "inverse": false,
                  "listItem": false,
                  "mask": null,
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": "2x",
                  "spin": false,
                  "symbol": false,
                  "title": "",
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    style={
                      Object {
                        "width": "undefinedpx",
                      }
                    }
                  />,
                  <span />,
                ],
                "className": "timer",
              },
              "ref": Object {
                "current": null,
              },
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "style": Object {
                      "width": "undefinedpx",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
